python/cudf_polars/cudf_polars/dsl/translate.py

```diff
@@ -796,7 +792,14 @@ def _(
     inner = translator.translate_expr(n=node.expr, schema=schema)
     # Push casts into literals so we can handle Cast(Literal(Null))
     if isinstance(inner, expr.Literal):
-        return expr.Literal(dtype, inner.value.cast(plc.interop.to_arrow(dtype)))
+        plc_column = plc.Column.from_scalar(
+            plc.Scalar.from_py(inner.value, inner.dtype), 1
+        )
+        casted_column = plc.unary.cast(plc_column, dtype)
+        casted_py_scalar = plc.interop.to_arrow(
+            plc.copying.get_element(casted_column, 0)
+        ).as_py()
```

Comment:
Maybe we need to support `Scalar.to_py` as well... how are we doing this in the legacy cudf.Scalar code paths (and the underlying code paths that we are keeping post the Scalar deprecation)? Is it all via `to_arrow`?

Reply:
> how are we doing this in the legacy cudf.Scalar code paths (and the underlying code paths that we are keeping post the Scalar deprecation)?

Yeah cuDF Python still largely does this though `pyarrow.Scalar`. What would be ideal is if we could cast scalars without going through column.

Reply:
Two orthogonal follow-ups then:
- We need to figure out what operations are not currently supported on Scalars that we need in libcudf.
- We need to determine whether pylibcudf.Scalar needs to be convertible to native host types where possible. I suspect that the answer is no since that won't be sufficient in the general case anyway (we need to support all Arrow types) and the only way to do that is by putting the Scalar into a column and then using the 0-copy interface since scalar layouts are not prescribed by the Arrow spec.

Neither of these needs to block this PR, though.